### ISSUE TITLE
Allow add_request_headers on Host resource

### ISSUE THREAD

**Issue by carlin-q-scott** [emissary-ingress/emissary#4854]
**Please describe your use case / problem.**
I want to map each Host to a specific tenant for downstream applications to be aware of.

**Describe the solution you'd like**
Allow setting add_request_headers on the Host resource.

**Describe alternatives you've considered**
I've already implemented a mapping in my AuthService that sets the tenant header based on the forwarded host header. 

I'm also considering using the kubernetes API from my AuthService to get metadata for the Host resources.

**Additional context**
Add any other context about the feature request here.

**Comment by cindymullins-dw** [comment]
Hi @carlin-q-scott , can you provide more context on why this feature is needed?  Add_request_headers is available on Mappings.  Do you have a Host with many Mappings where you'd want to apply the same add_request_headers for all?

**Comment by carlin-q-scott** [comment]
Yes, all my Mappings and AuthSevice are for all Hosts *. But each host is associated with a specific tenant.
